frontend/lib/src/components/widgets/ButtonGroup/ButtonGroup.tsx

```diff
@@ -0,0 +1,314 @@
+): React.FunctionComponent {
+  const isVisuallySelected = showAsSelected(
+    selectionVisualization,
+    clickMode,
+    selected,
+    index
+  )
+  const content = getContent(
+    isVisuallySelected,
+    option.content ?? "",
+    option.selectedContent
+  )
+
+  // we have to use forwardRef here becaused BasewebButtonGroup passes it down to its children
+  const buttonKind =
+    !isVisuallySelected || option.disableSelectionHighlight || false
+      ? BaseButtonKind.BORDERLESS_ICON
+      : BaseButtonKind.BORDERLESS_ICON_ACTIVE
+  return forwardRef(function BaseButtonGroup(
+    props: any,
+    _: Ref<BasewebButtonGroup>
+  ): ReactElement {
+    return (
+      <BaseButton {...props} size={BaseButtonSize.XSMALL} kind={buttonKind}>
+        {getContentElement(content)}
+      </BaseButton>
+    )
+  })
+}
+
+function getInitialValue(
+  widgetMgr: WidgetStateManager,
+  element: ButtonGroupProto
+): number[] {
+  const storedValue = widgetMgr.getIntArrayValue(element)
+  return storedValue ?? element.default
+}
+
+function ButtonGroup(props: Readonly<Props>): ReactElement {
+  const { disabled, element, fragmentId, widgetMgr } = props
+  const {
+    clickMode,
+    default: defaultValues,
+    options,
+    setValue,
+    value,
+    selectionVisualization,
+  } = element
+
+  const theme: EmotionTheme = useTheme()
+
+  const [selected, setSelected] = useState<number[]>(
+    getInitialValue(widgetMgr, element) || []
+  )
+
+  const elementRef = React.useRef(element)
+  // set to undefined for the first render so we know when its mounted
+  const selectedRef = React.useRef<number[] | undefined>(undefined)
+
+  // This is required for the form clearing functionality:
+  useEffect(() => {
+    if (!element.formId) {
+      // We don't need the form clear functionality if its not in a form
+      // or if selections are not activated.
+      return
+    }
+
+    const formClearHelper = new FormClearHelper()
+    // On form clear, reset the selections (in chart & widget state)
+    formClearHelper.manageFormClearListener(widgetMgr, element.formId, () => {
+      setSelected(defaultValues)
+    })
+
+    return () => {
+      formClearHelper.disconnect()
+    }
+  }, [element.formId, widgetMgr, defaultValues, element, fragmentId])
+
+  const valueString = JSON.stringify(value)
+  useEffect(() => {
+    const parsedValue = JSON.parse(valueString)
+    if (setValue) {
+      setSelected(parsedValue)
+      syncValue(selected, elementRef.current, widgetMgr, fragmentId, false)
+      elementRef.current.setValue = false
+    } else {
+      // only commit to the backend if the value has changed
+      if (isEqual(selected, selectedRef.current)) {
+        return
+      }
+      const fromUi = selectedRef.current === undefined ? false : true
+      syncValue(selected, elementRef.current, widgetMgr, fragmentId, fromUi)
+    }
+    selectedRef.current = selected
+  }, [selected, widgetMgr, fragmentId, valueString, setValue])
```

Comment:
I think `fragmentId` can be removed from dependencies? Also, I think `setValue` doesn't need to be added to dependencies, but maybe double check this.

Reply:
Hm I believe they have to stay in the deps array, because`fragmentId` is passed to the `syncValue` (now `syncWithWidgetManager`) method and `setValue` is used inside of `useEffect`. as well